lib/saluki-env/src/autodiscovery/providers/local.rs

```diff
@@ -0,0 +1,401 @@
+            service_id: String::new(),
+            tagger_entity: String::new(),
+            cluster_check: false,
+            node_name: String::new(),
+            source: String::new(),
+            ignore_autodiscovery_tags: false,
+            metrics_excluded: false,
+            logs_excluded: false,
+            advanced_ad_identifiers: Vec::new(),
+        };
+
+        let event = AutodiscoveryEvent { config };
+        broadcast_event(event_senders, event).await;
+    }
+
+    Ok(())
+}
+
+/// Broadcasts an event to all subscribers, removing any that have closed their channels
+async fn broadcast_event(senders: &Arc<Mutex<Vec<mpsc::Sender<AutodiscoveryEvent>>>>, event: AutodiscoveryEvent) {
+    let mut to_remove = Vec::new();
+
+    {
+        let senders_guard = senders.lock().unwrap();
+        for (idx, sender) in senders_guard.iter().enumerate() {
+            match sender.try_send(event.clone()) {
+                Ok(_) => {}
+                Err(TrySendError::Closed(_)) => {
+                    // Channel is closed, mark for removal
+                    to_remove.push(idx);
+                }
+                Err(TrySendError::Full(_)) => {
+                    // Channel is full, this is a backpressure situation
+                    warn!("Channel full - subscriber cannot keep up with autodiscovery events");
+                }
+            }
+        }
+    }
+
+    // Remove closed senders if any
+    if !to_remove.is_empty() {
+        let removed_count = to_remove.len();
+        let mut senders_guard = senders.lock().unwrap();
+        // Remove from back to front to avoid index shifting issues
+        to_remove.sort_unstable_by(|a, b| b.cmp(a));
+        debug!("Removing {} closed autodiscovery subscribers", removed_count);
+
+        for idx in to_remove {
+            senders_guard.swap_remove(idx);
+        }
+    }
+}
+
+#[async_trait]
+impl AutoDiscoveryProvider for LocalAutoDiscoveryProvider {
+    type Error = GenericError;
+
+    async fn subscribe(&mut self, sender: mpsc::Sender<AutodiscoveryEvent>) -> Result<(), Self::Error> {
+        {
+            let mut senders = self.event_senders.lock().unwrap();
+
+            // Check if this sender's address is already in our list
+            let sender_addr = format!("{:p}", &sender);
+            if senders.iter().any(|s| format!("{:p}", s) == sender_addr) {
+                debug!("Attempt to subscribe with duplicate sender - ignoring");
+                return Ok(());
+            }
+
+            senders.push(sender);
+        }
+
+        // Scan for initial configurations
+        let mut local_known = self.known_configs.clone();
+        scan_and_emit_events(&self.search_paths, &mut local_known, &self.event_senders).await?;
+        self.known_configs = local_known;
+
+        // Start background monitoring if not already active
+        if !self.monitoring_active {
+            self.start_background_monitor(30)?; // Check every 30 seconds
+        }
```

Comment:
The local implementation is not ideal in the sense that every subscription will trigger a broadcast to all the senders. We can probably change the implementation so `scan_and_emit_events` can accept a list of senders that way in the subscribe function we could pass a list with the newly added sender, then inside the `start_background_monitor` we can pass the full list of senders.